main.sh

```diff
@@ -144,21 +144,28 @@ EOF
 
 # Translate usage string -> getopts arguments, and set $arg_<flag> defaults
 while read -r __b3bp_tmp_line; do
-  if echo "${__b3bp_tmp_line}" |egrep '^-' >/dev/null 2>&1; then
+  if [[ "${__b3bp_tmp_line}" =~ ^- ]]; then
     # fetch single character version of option string
-    __b3bp_tmp_opt="$(echo "${__b3bp_tmp_line}" |awk '{print $1}' |sed -e 's#^-##')"
+    __b3bp_tmp_opt="${__b3bp_tmp_line%% *}"
+    __b3bp_tmp_opt="${__b3bp_tmp_opt:1}"
 
     # fetch long version if present
-    __b3bp_tmp_long_opt="$(echo "${__b3bp_tmp_line}" |awk '/\-\-/ {print $2}' |sed -e 's#^--##')"
-    __b3bp_tmp_long_opt_mangled="$(sed 's#-#_#g' <<< "$__b3bp_tmp_long_opt")"
+    __b3bp_tmp_long_opt=""
+
+    if [[ "${__b3bp_tmp_line}" =~ -- ]]; then
```

Comment:
but thinking about this line again made me realise that there is no need for regex in this case where a simple pattern match would suffice. updated & squashed the PR